### PR TITLE
Auto accept incoming federated shares from trusted servers

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
TODO:
- [ ] add UI
- [ ] Resolve cross app dependency somehow

## Related Issue
https://github.com/owncloud/core/issues/27653#issuecomment-380506261

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/federatedfilesharing/lib/RequestHandler.php [hunk 1]
```diff
@@ -214,9 +244,38 @@ public function createShare($params) {
 	}
 
 	/**
-	 * create re-share on behalf of another user
+	 * We can auto accept incoming shares if
+	 * - incoming shares do not add source to the trusted servers
+	 * - config option allows it
```
Comment (mmattel) [inline apps/federatedfilesharing/lib/RequestHandler.php]: Maybe to add that this is not a config.php option but a option inside the Federation app

**Comment by VicDeo** [comment]
@PVince81 well, this works but introduces a cross-app dependency between  `Federation` and `FederatedFileSharing` because FederatedFileSharing needs a trusted servers list that is managed by the federation.

**Comment by PVince81** [comment]
Hmm... maybe that's ok just like files_sharing and files_trashbin somehow depend on files...

**Comment by VicDeo** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by PVince81** [comment]
moving to "triage" for scheduling

**Comment by VicDeo** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by VicDeo** [comment]
Closed in favor of https://github.com/owncloud/core/pull/33980

**Pull request closed without merging**